src/library_dylink.js

```diff
@@ -637,9 +625,13 @@ var LibraryDylink = {
       }
 
       if (flags.loadAsync) {
-        return fetchBinary(libFile);
+        return new Promise(function(resolve, reject) {
+          readAsync(libFile, function(data) { resolve(new Uint8Array(data)); }, reject);
+        });
       }
+
       // load the binary synchronously
+      if (!readBinary) throw new Error('file not found');
```

Comment:
```suggestion
      if (!readBinary) {
        throw new Error(`file ${libfile} not available, and no way to read it synchronously`);
      }
```

Reply:
This is tricky one.  Previously, the user would just have see the "file not found" error from the filsystem and we would never have got here.    I will re-phrase.